src/components/Listing/ListingFilter.jsx

```diff
@@ -0,0 +1,167 @@
+
+import { useState } from "react";
+
+import { useSortedProduct } from "../../hooks/Filters/product-context";
+
+export const ProductFilter = () => {
+const {state,dispatch}=useSortedProduct();
+
+const [filterMenu, showFilterMenu] = useState(true)
+
+return (<div className="body-container">
+    <div className="side-bar">
+        {filterMenu ? <div className="filter-div-lg">
+            <div className="h4 filter-header">
+                <div>Filters</div>
+                <div className="clear-filter"
+                    onClick={() => dispatch({ type: "clear" })}
+                >Clear</div>
+            </div>
+            <div className="discount-percentage">
+                <div className="h5">Discount Price</div>
+
+                <input step="25-" type="range" min="0" max="1000"
+                    list="tickmarks"
+                    checked={state.discountValue}
+                    onClick={(e) => dispatch({ type: "discount", payload: e.target.value })}
+                />
+                <datalist id="tickmarks">
+                    <option value="0">0</option>
+                    <option value="25">250</option>
+                    <option value="50">500</option>
+                    <option value="75">750</option>
+                </datalist>
+                <label className="filter-price">
+                    <p id="price-max">0</p>
+                    <p id="price-min">250</p>
+                    <p id="price-mid">500</p>
+                    <p id="price-max">750</p>
+                </label>
+
+            </div>
+            <div className="category-selector">
+                <div className="h5">Sort Price</div>
+                <p>
+                    <input
+                        className="category"
+                        type="radio" id="price_sort"
+                        name="price_sort" value="price_sort"
+                        checked={state.priceHighLow === "low_to_high"}
+                        onClick={() => dispatch({ type: "price_sort", payload: "low_to_high" })}
```

Comment:
you can extract out these functions to the top of your component. This will result in clean code view.

Reply:
done . extracted the function to a new file.